### PR TITLE
feat(metador-team) signup page

### PR DESCRIPTION
**Pull request by AbuhaithemAlthry**
This is a simple mobile signup page designed to allow new users to create an account and access the app's features. The page is built with Flutter and includes basic validation for email and password fields.

To run the app, clone the repository and run 'flutter pub get' and 'flutter run'. Any feedback or suggestions for improvement are welcome. Thank you for reviewing my project!
image

### CODE REVIEW

File: Mobile/starter_project_matador/lib/features/example/presentation/screen/signup_page.dart [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import '../widgets/data_fill.dart';
+import '../widgets/white_box.dart';
+
+class SignUpPage extends StatelessWidget {
+  const SignUpPage({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        backgroundColor: Colors.white,
```
Comment (BekyH) [inline Mobile/starter_project_matador/lib/features/example/presentation/screen/signup_page.dart:14]: Avoid giving colors this way. There is a file where colors are defined inside the core/utils/constants directory. use that file to define colors and use them. This apply for all colors.

File: Mobile/starter_project_matador/lib/features/example/presentation/widgets/data_fill.dart [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        children: <Widget>[
+          Padding(
+            padding: EdgeInsets.fromLTRB(
+                (41.0 / 375) * MediaQuery.of(context).size.width,
+                (32.0 / 812) * MediaQuery.of(context).size.height,
+                0.0,
+                0.0),
+            child: Text(
+              'WELCOME',
+              style: TextStyle(
```
Comment (BekyH) [inline Mobile/starter_project_matador/lib/features/example/presentation/widgets/data_fill.dart:34]: Similar to the above comment. there are also styles defined inside the core/utils/constants directory. define the textstyles there and use them. This apply for all the styles.

File: Mobile/starter_project_matador/lib/features/example/presentation/widgets/signup_button.dart [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import "package:flutter/material.dart";
+
+class Button extends StatelessWidget {
```
Comment (BekyH) [inline Mobile/starter_project_matador/lib/features/example/presentation/widgets/signup_button.dart:3]: is there another better naming for this class? may be something like customButton or like that?

File: Mobile/starter_project_matador/lib/features/example/presentation/widgets/white_box.dart [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/material.dart';
+
+// This box is made inorder the UI fit the figma design
+class Box extends StatelessWidget {
```
Comment (BekyH) [inline Mobile/starter_project_matador/lib/features/example/presentation/widgets/white_box.dart:4]: Also customBox or sth else ? for class name?

**Review by BekyH — CHANGES_REQUESTED** [review]
Also don't modify example folder. migrate all your changes to features directory. sth like features/signup.

**Comment by BekyH** [comment]
> > Also don't modify example folder. migrate all your changes to features directory. sth like features/signup.
> 
> Actually I didn't touch other files rather I added mines

yes you added your files in example folder. just don't work inside example folder. create your own folder and move the file to that folder.

**Comment by BekyH** [comment]
> > Also don't modify example folder. migrate all your changes to features directory. sth like features/signup.
> 
> Actually I didn't touch other files rather I added mines

yes you added your files in example folder. just don't work inside example folder. create your own folder and move the file to that folder.

File: Mobile/starter_project_matador/lib/features/example/presentation/widgets/data_fill.dart (hunk 2 above)
Comment (AbuhaithemAlthry) [reply Mobile/starter_project_matador/lib/features/example/presentation/widgets/data_fill.dart:31]: I have resolved this problem

File: Mobile/starter_project_matador/lib/features/example/presentation/screen/signup_page.dart (hunk 1 above)
Comment (AbuhaithemAlthry) [reply Mobile/starter_project_matador/lib/features/example/presentation/screen/signup_page.dart:16]: done avoiding it

File: Mobile/starter_project_matador/lib/features/example/presentation/widgets/signup_button.dart (hunk 3 above)
Comment (AbuhaithemAlthry) [reply Mobile/starter_project_matador/lib/features/example/presentation/widgets/signup_button.dart:3]: done resolving it

**Review by BekyH — DISMISSED** [review]

File: Mobile/starter_project_matador/lib/features/example/presentation/widgets/signup_and_signin_box.dart [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+import '../../../../core/utils/constants/styles.dart';
+
+class SigninSignupBox extends StatelessWidget {
+  const SigninSignupBox({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    var Height = MediaQuery.of(context).size.height;
+    var Width = MediaQuery.of(context).size.width;
+    var row = Align(
```
Comment (Redi22) [inline Mobile/starter_project_matador/lib/features/example/presentation/widgets/signup_and_signin_box.dart:12]: Why? this should either be its own widget or merged with the current widget.

**Review by Redi22 — CHANGES_REQUESTED** [review]

File: Mobile/starter_project_matador/lib/features/example/presentation/widgets/signup_and_signin_box.dart (hunk 5 above)
Comment (AbuhaithemAlthry) [reply Mobile/starter_project_matador/lib/features/example/presentation/widgets/signup_and_signin_box.dart]: I have corrected this one

**Review by BekyH — DISMISSED** [review]

**Review by BekyH — APPROVED** [review]

**Pull request closed without merging**